pkg/controller/stack/stack_controller.go

```diff
@@ -0,0 +1,981 @@
+	err = c.Watch(&source.Kind{Type: &pipelinev1alpha1.Pipeline{}}, tH, tPred)
+	if err != nil {
+		log.Info(fmt.Sprintf("Tekton Pipelines may not be installed"))
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &pipelinev1alpha1.Task{}}, tH, tPred)
+	if err != nil {
+		log.Info(fmt.Sprintf("Tekton Pipelines may not be installed"))
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileStack implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileStack{}
+
+// ReconcileStack reconciles a Stack object
+type ReconcileStack struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+
+	//The indexResolver which will be used during reconciliation
+	indexResolver func(kabanerov1alpha2.RepositoryConfig) (*Index, error)
+}
+
+// Reconcile reads that state of the cluster for a Stack object and makes changes based on the state read
+// and what is in the Stack.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileStack) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx := context.Background()
+
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Stack")
+
+	// Fetch the Stack instance
+	instance := &kabanerov1alpha2.Stack{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Resolve the kabanero instance
+	// TODO: issue #92, when repo is added to the Stack, there will be no need for the Kabanero
+	//       object here.
+	var k *kabanerov1alpha2.Kabanero
+	l := kabanerov1alpha2.KabaneroList{}
+	err = r.client.List(context.Background(), &l, client.InNamespace(instance.GetNamespace()))
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	for _, _k := range l.Items {
+		k = &_k
+	}
+	reqLogger.Info("Resolved Kabanero", "kabanero", k)
+
+	// If the stack is being deleted, and our finalizer is set, process it.
+	beingDeleted, err := processDeletion(ctx, instance, r.client, reqLogger)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if beingDeleted {
+		return reconcile.Result{}, nil
+	}
+
+	rr, err := r.ReconcileStack(instance, k)
```

Comment:
Hopefully do not need to pass kabanero instance to this method anymore, see later comments...